tests/out/interpolate.Fragment.glsl

```diff
@@ -0,0 +1,28 @@
+#version 310 es
+
+precision highp float;
+
+struct FragmentInput {
+    vec4 position;
+    uint flat1;
+    float linear;
+    vec2 linear_centroid;
+    vec3 linear_sample;
+    vec4 perspective;
+    float perspective_centroid;
+    float perspective_sample;
+};
+
+flat layout(location = 0) in uint _vs2fs_location0;
+layout(location = 1) in float _vs2fs_location1;
+layout(location = 2) centroid in vec2 _vs2fs_location2;
+layout(location = 3) in vec3 _vs2fs_location3;
```

Comment:
hmm, why is there no `noperspective` on any of them?

Reply:
The version of this PR rebased on #722 has `noperspective` as required. The omission here was caused by the initial bogus behavior where GLSL output just omitted qualifiers not supported in the requested version. With #722, requesting unsupported qualifiers causes an error, and the snapshot tests request a GLSL version that does support `noperspective`.